commons/src/lib.rs

```diff
@@ -1,15 +1,93 @@
+#[macro_use]
+extern crate failure;
+extern crate url;
+
+use failure::Fallible;
+use std::collections::HashSet;
+use url::form_urlencoded;
+
 /// Strip all but one leading slash and all trailing slashes
 pub fn parse_path_prefix(path_prefix: &str) -> String {
     format!("/{}", path_prefix.to_string().trim_matches('/'))
 }
 
+/// Parse a comma-separated set of client parameters keys.
+pub fn parse_params_set(params: &str) -> HashSet<String> {
+    params
+        .split(',')
+        .into_iter()
```

Comment:
This line is redundant as [split already returns an iterator](https://doc.rust-lang.org/std/primitive.str.html#iterator-behavior)

Reply:
Oh, nice!